FinSearchUnified/Tests/Bundle/SearchBundleFindologic/ResponseParser/Xml21ResponseParserTest.php

```diff
@@ -0,0 +1,229 @@
+        $expectedLink
+    ) {
+        $response = Utility::getDemoResponse($filename);
+        $responseParser = ResponseParser::getInstance($response);
+        $smartDidYouMean = $responseParser->getSmartDidYouMean();
+
+        $this->assertInstanceOf(SmartDidYouMean::class, $smartDidYouMean);
+        $this->assertSame($expectedType, $smartDidYouMean->getType());
+        $this->assertSame($expectedLink, $smartDidYouMean->getLink());
+    }
+
+    public function testProductsWhenResponseDoesNotHaveProducts()
+    {
+        $response = Utility::getDemoResponse('demoResponseWithNoResults.xml');
+        $responseParser = ResponseParser::getInstance($response);
+        $products = $responseParser->getProducts();
+        $this->assertEmpty($products);
+    }
+
+    public function testProductsWhenResponseContainsOneProduct()
+    {
+        $article = Utility::createTestProduct(1, true);
+        $xmlResponse = Utility::getDemoXML('demoResponseWithOneProduct.xml');
+        $xmlResponse->products->product[0]->attributes()->id = $article->getId();
+
+        $response = new Xml21Response($xmlResponse->asXML());
+        $responseParser = ResponseParser::getInstance($response);
+        $products = $responseParser->getProducts();
+        $this->assertCount(1, $products);
+
+        foreach ($products as $product) {
+            $this->assertEquals($product['orderNumber'], $article->getMainDetail()->getNumber());
+            $this->assertEquals($product['detailId'], $article->getMainDetail()->getId());
+        }
+    }
+
+    public function testProductsWhenResponseContainsOneProductThatDoesNotExist()
+    {
+        $response = Utility::getDemoResponse('demoResponseWithOneProduct.xml');
+        $responseParser = ResponseParser::getInstance($response);
+        $products = $responseParser->getProducts();
+        $this->assertEmpty($products);
+    }
+
+    public function testResponseWhenLandingPageExists()
+    {
+        $response = Utility::getDemoResponse('demoResponseWithLandingPage.xml');
+        $responseParser = ResponseParser::getInstance($response);
+        $uri = $responseParser->getLandingPageUri();
+        $this->assertNotNull($uri);
+        $this->assertSame('https://blubbergurken.io', $uri);
+    }
+
+    public function testResponseWhenLandingPageDoesNotExist()
+    {
+        $response = Utility::getDemoResponse('demoResponse.xml');
+        $responseParser = ResponseParser::getInstance($response);
+        $uri = $responseParser->getLandingPageUri();
+        $this->assertNull($uri);
+    }
+
+    public function testResponseWhenPromotionDoesNotExist()
+    {
+        $response = Utility::getDemoResponse('demoResponse.xml');
+        $responseParser = ResponseParser::getInstance($response);
+        $promotion = $responseParser->getPromotion();
+        $this->assertNull($promotion);
+    }
+
+    public function testResponseWhenPromotionExists()
+    {
+        $response = Utility::getDemoResponse('demo.xml');
```

Comment:
Why are you using `demo.xml` instead of `demoResponse.xml`? Based on the name they should have the same contents. If that's not the case, the name of the file should reflect the response contents.